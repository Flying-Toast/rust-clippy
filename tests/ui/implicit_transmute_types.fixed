//@run-rustfix
#![allow(unused)]
#![warn(clippy::implicit_transmute_types)]

use std::mem;

fn main() {
    unsafe {
        let _: u32 = mem::transmute::<i32, u32>(123i32);

        let _: u32 = mem::transmute::<i32, u32>(123i32);
        let _: u32 = mem::transmute::<_, _>(123i32);
    }
}
