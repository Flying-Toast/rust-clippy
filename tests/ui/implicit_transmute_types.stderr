error: `transmute` called without explicit type parameters
  --> $DIR/implicit_transmute_types.rs:9:22
   |
LL |         let _: u32 = mem::transmute(123i32);
   |                      ^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: `-D clippy::implicit-transmute-types` implied by `-D warnings`
help: consider specifying the types intended to be transmuted
   |
LL |         let _: u32 = mem::transmute::<i32, u32>(123i32);
   |                                    ++++++++++++

error: aborting due to previous error

